tests/unit/helm/api-deployment_test.go

```diff
@@ -217,6 +217,165 @@ func (s *deploymentApiTemplateTest) TestReplicas() {
 	}
 }
 
+func (s *deploymentApiTemplateTest) TestTopologySpreadConstraints() {
+	testCases := []struct {
+		name     string
+		values   map[string]string
+		expected func(constraint []corev1.TopologySpreadConstraint)
+	}{
+		{
+			"defaultValues",
+			nil,
+			func(constraint []corev1.TopologySpreadConstraint) {
+				var expectedTopologySpreadConstraint []corev1.TopologySpreadConstraint
+				s.Equal(expectedTopologySpreadConstraint, constraint, "Constraints should be equal")
+			},
+		},
+		{
+			"overrideTopologySpreadConstraintsRequiredValues",
+			map[string]string{
+				"apiSettings.topologySpreadConstraints[0].maxSkew":           "1",
+				"apiSettings.topologySpreadConstraints[0].topologyKey":       "kubernetes.io/hostname",
+				"apiSettings.topologySpreadConstraints[0].whenUnsatisfiable": "DoNotSchedule",
+			},
+			func(constraint []corev1.TopologySpreadConstraint) {
+				var expectedTopologySpreadConstraint []corev1.TopologySpreadConstraint
+				eexpectedTopologySpreadConstraintJSON := `[
+					{
+					  "maxSkew": 1,
+					  "topologyKey": "kubernetes.io/hostname",
+					  "whenUnsatisfiable": "DoNotSchedule",
+					  "labelSelector": {
+					  	"matchLabels": {
+							"app.kubernetes.io/name": "teams-api",
+							"app.kubernetes.io/instance": "fiftyone-test"
+						}
+					  }
+					}
+				  ]`
+				err := json.Unmarshal([]byte(eexpectedTopologySpreadConstraintJSON), &expectedTopologySpreadConstraint)
+				s.NoError(err)
+				s.Equal(expectedTopologySpreadConstraint, constraint, "Constraints should be equal")
+			},
+		},
+		{
+			"overrideTopologySpreadConstraintsOptionalValues",
+			map[string]string{
+				"apiSettings.topologySpreadConstraints[0].matchLabelKeys":     "[\"pod-template-hash\"]",
+				"apiSettings.topologySpreadConstraints[0].maxSkew":            "1",
+				"apiSettings.topologySpreadConstraints[0].minDomains":         "1",
+				"apiSettings.topologySpreadConstraints[0].nodeAffinityPolicy": "Honor",
+				"apiSettings.topologySpreadConstraints[0].nodeTaintsPolicy":   "Honor",
+				"apiSettings.topologySpreadConstraints[0].topologyKey":        "kubernetes.io/hostname",
+				"apiSettings.topologySpreadConstraints[0].whenUnsatisfiable":  "DoNotSchedule",
+				"apiSettings.topologySpreadConstraints[1].matchLabelKeys":     "[\"pod-template-hash\"]",
+				"apiSettings.topologySpreadConstraints[1].maxSkew":            "1",
+				"apiSettings.topologySpreadConstraints[1].minDomains":         "1",
+				"apiSettings.topologySpreadConstraints[1].nodeAffinityPolicy": "Honor",
+				"apiSettings.topologySpreadConstraints[1].nodeTaintsPolicy":   "Honor",
+				"apiSettings.topologySpreadConstraints[1].topologyKey":        "kubernetes.io/hostname",
+				"apiSettings.topologySpreadConstraints[1].whenUnsatisfiable":  "DoNotSchedule",
```

Comment:
Maybe instead of duplicating the values from `apiSettings.topologySpreadConstraints[0].*` we could set some or all different to ensure that the proper element is used?

Reply:
Made `[1]` unique across the tests